x-pack/plugins/ml/public/jobs/jobs_list_new/components/jobs_list_view/jobs_list_view.js

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import './styles/main.less';
+
+import { ml } from 'plugins/ml/services/ml_api_service';
+import { loadFullJob } from '../utils';
+import { JobsList } from '../jobs_list';
+import { JobDetails } from '../job_details';
+import { EditJobFlyout } from '../edit_job_flyout';
+import { DeleteJobModal } from '../delete_job_modal';
+import { StartDatafeedModal } from '../start_datafeed_modal';
+import { MultiJobActions } from '../multi_job_actions';
+
+import React, {
+  Component
+} from 'react';
+
+export class JobsListView extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      jobsSummaryList: [],
+      fullJobsList: {},
+      selectedJobs: [],
+      itemIdToExpandedRowMap: {}
+    };
+
+    this.updateFunctions = {};
+
+    this.showEditJobFlyout = () => {};
+    this.showDeleteJobModal = () => {};
+    this.showStartDatafeedModal = () => {};
+
+    this.blockAutoRefresh = false;
+
+    this.refreshJobSummaryList();
+  }
+
+  componentWillUnmount() {
+    this.blockAutoRefresh = true;
+  }
+
+  toggleRow = (jobId) => {
+    if (this.state.itemIdToExpandedRowMap[jobId]) {
+      const itemIdToExpandedRowMap = { ...this.state.itemIdToExpandedRowMap };
+      delete itemIdToExpandedRowMap[jobId];
+      this.setState({ itemIdToExpandedRowMap });
+    } else {
+
+      let itemIdToExpandedRowMap = { ...this.state.itemIdToExpandedRowMap };
+
+      if (this.state.fullJobsList[jobId] !== undefined) {
+        itemIdToExpandedRowMap[jobId] = (
+          <JobDetails
+            jobId={jobId}
+            job={this.state.fullJobsList[jobId]}
+            addYourself={this.addUpdateFunction}
+          />
+        );
+      } else {
+        itemIdToExpandedRowMap[jobId] = (
+          <JobDetails
+            jobId={jobId}
+            addYourself={this.addUpdateFunction}
+          />
+        );
+      }
+
+      this.setState({ itemIdToExpandedRowMap }, () => {
+        loadFullJob(jobId)
+          .then((job) => {
+            const fullJobsList = { ...this.state.fullJobsList };
+            fullJobsList[jobId] = job;
+            this.setState({ fullJobsList }, () => {
+              // take a fresh copy of the itemIdToExpandedRowMap object
+              itemIdToExpandedRowMap = { ...this.state.itemIdToExpandedRowMap };
+              if (itemIdToExpandedRowMap[jobId] !== undefined) {
+                // wrap in a check, in case the user closes the expansion before the
+                // loading has finished
+                itemIdToExpandedRowMap[jobId] = (
+                  <JobDetails
+                    jobId={jobId}
+                    job={job}
+                    addYourself={this.addUpdateFunction}
```

Comment:
Might be worth an extra comment explaining what this `addYourself` and `removeYourself` functions are doing.